internal/configuration/configuration.go

```diff
@@ -131,18 +131,31 @@ func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
 		return nil
 	}
 
-	log.Tracef("updating configuration: %v", message)
 	var errors error
+
+	newJson, err := json.Marshal(message)
+	if err != nil {
+		errors = multierror.Append(errors, fmt.Errorf("cannot marshal message to JSON: %s", err))
+		return errors
+	}
+
+	oldJson, err := json.Marshal(m.deviceConfiguration)
+	if err != nil {
+		errors = multierror.Append(errors, fmt.Errorf("cannot marshal message to JSON: %s", err))
+		return errors
+	}
+
+	log.Infof("updating configuration. New config: %s\nOld config: %s", newJson, oldJson)
 	for _, observer := range m.observers {
 		err := observer.Update(message)
 		if err != nil {
 			errors = multierror.Append(errors, fmt.Errorf("running config observer failed: %s", err))
 		}
 	}
 
-	file, err := json.MarshalIndent(message, "", " ")
+	file, err := json.MarshalIndent(m.deviceConfiguration, "", " ")
```

Comment:
since it was already marshalled, you can only call `json.Indent` and save the need for marshalling once again.

Reply:
in addition, why to change it to `m.deviceConfiguration` ? it's the old configuration, no?

Reply:
fixed